xoa/data_source_host.go

```diff
@@ -57,6 +70,23 @@ func dataSourceHostRead(d *schema.ResourceData, m interface{}) error {
 	}
 
 	d.SetId(hosts[0].Id)
-
+	d.Set("pool_id", hosts[0].Pool)
+	d.Set("memory", hosts[0].Memory.Size)
+	d.Set("memory_usage", hosts[0].Memory.Usage)
+	d.Set("cpus", hostCpuInfoToMapList(hosts)[0])
+	d.Set("tags", hosts[0].Tags)
 	return nil
 }
+
+func hostCpuInfoToMapList(hosts []client.Host) []map[string]interface{} {
```

Comment:
This code isn't taking advantage of passing in a slice of `client.Host`s. I assume this was made to take a slice because there was hope that this would have worked for the hosts data source? Since it's only in use for the host data source, why don't we change the function signature to the following:

```
func hostCpuInfoToMapList(host client.Host) map[string]interface{}
```

Then the hosts data source can use it in its loop as well rather than duplicating this logic.